Sources/TokamakCore/Animation/_AnimationBoxBase.swift

```diff
@@ -0,0 +1,182 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 7/11/21.
+//
+
+import Foundation
+
+public class _AnimationBoxBase: Equatable {
+  public struct _Resolved {
+    public var duration: Double {
+      switch style {
+      case let .easeIn(duration),
+           let .easeOut(duration),
+           let .easeInOut(duration),
+           let .timingCurve(_, _, _, _, duration):
+        return duration
+      case let .solver(solver):
+        return solver.restingPoint(precision: 0.01)
+      }
+    }
+
+    public var delay: Double
+    public var speed: Double
+    public var repeatStyle: _RepeatStyle
+    public var style: _Style
+
+    public enum _Style: Equatable {
+      case easeIn(duration: Double),
+           easeOut(duration: Double),
+           easeInOut(duration: Double)
+      case timingCurve(Double, Double, Double, Double, duration: Double)
+      case solver(_AnimationSolver)
+
+      public static func == (lhs: Self, rhs: Self) -> Bool {
+        switch lhs {
+        case let .easeIn(lhsDuration):
+          if case let .easeIn(rhsDuration) = rhs {
+            return lhsDuration == rhsDuration
+          }
+        case let .easeOut(lhsDuration):
+          if case let .easeOut(rhsDuration) = rhs {
+            return lhsDuration == rhsDuration
+          }
+        case let .easeInOut(lhsDuration):
+          if case let .easeInOut(rhsDuration) = rhs {
+            return lhsDuration == rhsDuration
+          }
+        case let .timingCurve(lhs0, lhs1, lhs2, lhs3, lhsDuration):
+          if case let .timingCurve(rhs0, rhs1, rhs2, rhs3, rhsDuration) = rhs {
+            return lhs0 == rhs0
+              && lhs1 == rhs1
+              && lhs2 == rhs2
+              && lhs3 == rhs3
+              && lhsDuration == rhsDuration
+          }
+        case let .solver(lhsSolver):
+          if case let .solver(rhsSolver) = rhs {
+            return type(of: lhsSolver) == type(of: rhsSolver)
+          }
+        }
+        return false
+      }
+    }
+
+    public enum _RepeatStyle: Equatable {
+      case fixed(Int, autoreverses: Bool)
+      case forever(autoreverses: Bool)
+
+      public var autoreverses: Bool {
+        switch self {
+        case let .fixed(_, autoreverses),
+             let .forever(autoreverses):
+          return autoreverses
+        }
+      }
+    }
+  }
+
+  func resolve() -> _Resolved {
+    fatalError("implement \(#function) in subclass")
+  }
+
+  func equals(_ other: _AnimationBoxBase) -> Bool {
+    fatalError("implement \(#function) in subclass")
+  }
+
+  public static func == (lhs: _AnimationBoxBase, rhs: _AnimationBoxBase) -> Bool {
+    lhs.equals(rhs)
+  }
+}
+
+class StyleAnimationBox: _AnimationBoxBase {
```

Comment:
nit: would `final` make sense here?

```suggestion
final class StyleAnimationBox: _AnimationBoxBase {
```